### PR TITLE
Add Serialize/Deserialize to resource and stat types

### LINKED ISSUE #54: Add optional serde support for statistics structs
**Is your feature request related to a problem? Please describe.**

Gathering statistics via cgroups is great. They cover hierarchies of processes without the need to scan a list of task etc. The statistics contains a huge amount of values. Copying the values to a custom struct or merging them into a hashmap is manual work. Using the exact same structure and names as in the cgroup files has the advantage that the values are also documented. 
The structs defined for e.g `memory` or `blkio` do not derive from Serdes `Serialize` and `Deserialize` which would allow to directly serialise stats generated with e.g `blkio::BlkIoController::blkio`.

**Describe the solution you'd like**

Add an optional crate dependency `serde` with feature `derive`:

```toml
serde = { version = "1.0", features = ["derive"], optional = true }
```

Derive the statistics structs from `serde::Serialize` and `serde::Deserialize` if the feature `serde` is enabled e.g:

```rust
/// State of and statistics gathered by the kernel about the memory usage of the control group's
/// tasks.
#[derive(Debug)]
#[cfg_attr(feature="serde", derive(serde::Serialize, serde::Deserialize))]
pub struct Memory {
    /// How many times the limit has been hit.
    pub fail_cnt: u64,
    /// The limit in bytes of the memory usage of the control group's tasks.
    pub limit_in_bytes: i64,
    /// The current usage of memory by the control group's tasks.
    pub usage_in_bytes: u64,
...
```

See https://github.com/serde-rs/serde/issues/1021.

**Describe alternatives you've considered**

Following [Deriving De/Serialize for type in a different crate](https://serde.rs/remote-derive.html)

**Additional context**

A PR should be almost non invasive, because without the feature enabled there shouldn't be any change.

### LINKED ISSUE #55: Lift `'static` requirement on resource `attr` fields
**Which feature do you think can be improved?**

The key type of the additional attributes map of various resources like [MemoryResources](https://docs.rs/cgroups-rs/0.2.6/cgroups_rs/struct.MemoryResources.html) has the `'static` requirement. This is nice for manual setup but really hard for configurations read from files and applied later on. 

**How can it be improved?**

Either assign a dedicated lifetime to the structs that must be fulfilled or change the key type of the maps to `String` for the sake of simplicity.

**Additional Information**

Anything else to add? Just a big thank you for this crate ;-)

### PR DESCRIPTION
**Pull request by flxo**
Add feature "serde" that derives all resource and statistics types from Serialize and Deserialize. The feature is turned off by default.

Fixes #54
Fixes #55
Fixes #56

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @flxo !

lgtm

**Comment by flxo** [comment]
@jodh-intel Thanks for the review. Can we trigger the CI?

File: Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,10 @@ log = "0.4"
 regex = "1.1"
 nix = "0.20.0"
 libc = "0.2"
+serde = { version = "1.0", features = ["derive"], optional = true }
 
 [dev-dependencies]
 libc = "0.2.76"
+
+[features]
+default = ["serde"]
```
Comment (Tim-Zhang) [inline Cargo.toml:25]: enabled by default? but you have said: "The feature is turned off by default" in the description.

**Comment by Tim-Zhang** [comment]
@flxo Thanks for you pull reqeust, I have left a comment and have triggered the CI. the commit message check failed because The sign-off is missing. We usually add the sign-off by using `git commit -s` Thank you

**Comment by Tim-Zhang** [comment]
@flxo https://github.com/kata-containers/cgroups-rs/pull/56#discussion_r680617215 ?

File: Cargo.toml (hunk 1 above)
Comment (flxo) [reply Cargo.toml]: Fixed. Thanks. I removed the default feature `serde`.

File: src/lib.rs [hunk 2]
```diff
@@ -488,14 +492,15 @@ pub struct CpuResources {
     /// # Usage:
     /// ```
     /// let resource = &mut cgroups_rs::Resources::default();
-    /// resource.cpu.attrs.insert("cpu.cfs_init_buffer_us", "10".to_string());
+    /// resource.cpu.attrs.insert("cpu.cfs_init_buffer_us".to_string(), "10".to_string());
     /// // apply here
     /// ```
-    pub attrs: std::collections::HashMap<&'static str, String>,
+    pub attrs: HashMap<String, String>,
```
Comment (Tim-Zhang) [inline src/lib.rs:498]: API changed, need a major version bump.

Comment (flxo) [reply src/lib.rs:498]: Shouldn't for the major version `0` a bump of the minor or patch version fine?

**Comment by Tim-Zhang** [comment]
@flxo commit message check failed agian

**Comment by Tim-Zhang** [comment]
@flxo No need to bump the version, I'll do it after all pr merged.

**Review by lifupan — APPROVED** [review]

**Pull request merged**